python/src/aac/parser.py

```diff
@@ -8,11 +8,105 @@
 import os
 
 import yaml
+
 from attr import Factory, attrib, attrs, validators
 from yaml.parser import ParserError as YAMLParserError
 
+from aac.util import is_path_name
+
+
+@attrs
+class SourceLocation:
+    """The position and ... of an AaC structure in the source.
+
+    Attributes:
+        line (int): The line number on which the object was found.
+        column (int): The character position at which the object was found.
+        position (int): The position relative to the start of the file where the object was found.
+        span (int): The number of characters occupied by the object relative to `position`.
+    """
+
+    line: int = attrib(validator=validators.instance_of(int))
+    column: int = attrib(validator=validators.instance_of(int))
+    position: int = attrib(validator=validators.instance_of(int))
+    span: int = attrib(validator=validators.instance_of(int))
+
+
+@attrs
+class Lexeme:
+    """A lexical unit for a parsed AaC object.
+
+    Attributes:
+        location (SourceLocation): The location at which the object was found.
+        source (str): The source in which the object was found.
+        value (str): The value of the parsed object.
+    """
+
+    location: SourceLocation = attrib(validator=validators.instance_of(SourceLocation))
+    source: str = attrib(validator=validators.instance_of(str))
+    value: str = attrib(validator=validators.instance_of(str))
+
+
+@attrs
+class ParsedModel:
```

Comment:
So, this is circling back to our conversation about terminology. If we're wanting to use `definition` to describe all possible AaC entries and let `model` describe those definitions with `model:` as their root key, then we should change this name for clarity. 
```suggestion
class ParsedDefinition:
```

Reply:
There are obviously numerous references throughout that will need to be changed, but I don't want to nit-pick them until we align on the naming above.